src/makeCalculator.js

```diff
@@ -37,7 +37,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    add(a, b) {
+      calculator.result = a + b;
+
+      return calculator.result;
+    },
+    subtract(a, b) {
+      calculator.result = a - b;
+
+      return calculator.result;
+    },
+    multiply(a, b) {
+      calculator.result = a * b;
+
+      return calculator.result;
+    },
+    divide(a, b) {
+      calculator.result = a / b;
+
+      return calculator.result;
+    },
+    reset() {
+      calculator.result = 0;
+
+      return calculator;
+    },
+    operate(callback, num) {
+      calculator.result = callback(calculator.result, num);
```

Comment:
You could use `callback.call(this, num);` when invoking methods like `add` inside `operation`
This gives context to the methods called by `operation` and lets them use `this` in reference to the full `calculator` object inside each of its methods, for example: `this.result = 0;` for the reset method. We are not taught `.call` or other object methods before this task, so I'm marking this solution as approved.

Reply:
yup, I agree 😄